dashboard/models.py

```diff
@@ -212,24 +211,29 @@ class Course(models.Model):
     def __str__(self):
         return self.name
 
-    @property
-    def course_date_range(self):
+    def determine_date_start(self) -> datetime:
         if self.date_start is not None:
             start = self.date_start
         elif self.term is not None and self.term.date_start is not None:
             start = self.term.date_start
         else:
             logger.info(f"No date_start value was found for course {self.name} ({self.canvas_id}) or term; setting to current date and time")
             start = datetime.now(pytz.UTC)
+        return start
+
+    def determine_date_end(self, start: Union[datetime, None] = None) -> datetime:
```

Comment:
Maybe passing the start here is too fancy, but maybe it keeps things more consistent and saves a query in some circumstances.
Scenarios to consider:
- No course dates, no term dates, current time and two weeks from now; better synchronized but probably only by milliseconds, which is basically irrelevant.
- No date end, no term dates, but a course start date. Query for course doesn't need to be re-made when getting the course start?

These may not matter much anyway, since I tried accessing a course without a term (which I manually deleted), and it said that course data was still being processed.

Reply:
I was able to test these scenarios and I was able to get correct result.

In case of no end date, Things were fine as well. I don't think Assignment planning relay on end date. But resource access view does and in that case the bar was showing 2 weeks only. The behavior is same from before. All this change was doing is name change of the  function.